src/hevm/hevm-cli/hevm-cli.hs

```diff
@@ -593,24 +596,32 @@ dappCoverage opts _ solcFile =
         let
           dapp = dappInfo "." contractMap sourceCache
           f (k, vs) = do
-            putStr ("\x1b[0m" ++ "————— hevm coverage for ") -- Prefixed with color reset
-            putStrLn (unpack k ++ " —————")
-            putStrLn ""
-            forM_ vs $ \(n, bs) -> do
-              case ByteString.find (\x -> x /= 0x9 && x /= 0x20 && x /= 0x7d) bs of
-                Nothing -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
-                Just _ ->
-                  case n of
-                    -1 -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
-                    0  -> putStr "\x1b[31m" -- Red (Uncovered)
-                    _  -> putStr "\x1b[32m" -- Green (Covered)
-              Char8.putStrLn bs
-            putStrLn ""
-
+            if shouldPrintCoverage (EVM.UnitTest.covMatch opts) k then do
+              putStr ("\x1b[0m" ++ "————— hevm coverage for ") -- Prefixed with color reset
+              putStrLn (unpack k ++ " —————")
+              putStrLn ""
+              forM_ vs $ \(n, bs) -> do
+                case ByteString.find (\x -> x /= 0x9 && x /= 0x20 && x /= 0x7d) bs of
+                  Nothing -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
+                  Just _ ->
+                    case n of
+                      -1 -> putStr "\x1b[38;5;240m" -- Gray (Coverage status isn't relevant)
+                      0  -> putStr "\x1b[31m" -- Red (Uncovered)
+                      _  -> putStr "\x1b[32m" -- Green (Covered)
+                Char8.putStrLn bs
+              putStrLn ""
+            else pure ()
         mapM_ f (Map.toList (coverageReport dapp covs))
       Nothing ->
         error ("Failed to read Solidity JSON for `" ++ solcFile ++ "'")
 
+shouldPrintCoverage :: Maybe Text -> Text -> Bool
+shouldPrintCoverage (Just covMatch) name = regexMatches covMatch name
+shouldPrintCoverage Nothing name = not (isCoverageIrrelevant name)
+
+isCoverageIrrelevant :: Text -> Bool
+isCoverageIrrelevant name = Text.isPrefixOf "src/external/" name || Text.isPrefixOf "src/tests/" name || Text.isPrefixOf "src/test/" name || Text.isPrefixOf "lib/" name || Text.isSuffixOf ".t.sol" name
```

Comment:
Is there a more haskell-y way to write this? Maybe a function that takes an array of text called `areAnyPrefixOf`?